modules/geochemistry/src/outputs/GeochemistryConsoleOutput.C

```diff
@@ -0,0 +1,318 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "GeochemistryConsoleOutput.h"
+#include "GeochemistryConstants.h"
+#include "GeochemistryFormattedOutput.h"
+#include "GeochemistrySortedIndices.h"
+
+registerMooseObject("GeochemistryApp", GeochemistryConsoleOutput);
+
+InputParameters
+GeochemistryConsoleOutput::validParams()
+{
+  InputParameters params = Output::validParams();
+  params.addRequiredParam<UserObjectName>("geochemistry_reactor",
+                                          "The name of the GeochemistryReactor UserObject");
+  params.addParam<unsigned int>("precision", 4, "Precision for printing values");
+  params.addParam<Real>(
+      "mol_cutoff",
+      1E-40,
+      "Information regarding species with molalities less than this amount will not be outputted");
+  params.addRangeCheckedParam<Real>("stoichiometry_tolerance",
+                                    1E-6,
+                                    "stoichiometry_tolerance >= 0.0",
+                                    "if abs(any stoichiometric coefficient) < stoi_tol then it is "
+                                    "set to zero, and so will not appear in the output");
+  params.addParam<bool>(
+      "solver_info", false, "Print information from the solver including residuals, swaps, etc");
+  params.addRequiredParam<Point>("point",
+                                 "The physical point at which to query the GeochemistryReactor");
+  params.addClassDescription("Outputs results from a GeochemistryReactor at a particular point");
+  return params;
+}
+
+GeochemistryConsoleOutput::GeochemistryConsoleOutput(const InputParameters & parameters)
+  : Output(parameters),
+    UserObjectInterface(this),
+    _reactor(getUserObject<GeochemistryReactorBase>("geochemistry_reactor")),
+    _point(getParam<Point>("point")),
+    _precision(getParam<unsigned int>("precision")),
+    _stoi_tol(getParam<Real>("stoichiometry_tolerance")),
+    _solver_info(getParam<bool>("solver_info")),
+    _mol_cutoff(getParam<Real>("mol_cutoff"))
+{
+}
+
+void
+GeochemistryConsoleOutput::output(const ExecFlagType & type)
```

Comment:
Not necessary to attack in this PR,  but we do have some great table output utilities that might prettify your output some :)

Reply:
Oh, that might be nice.  Let's leave it for now, probably, but can you show me an example?